packages/ember-htmlbars/tests/integration/component_invocation_test.js

```diff
@@ -595,367 +66,14 @@ QUnit.test('moduleName is available on _renderNode when no layout is present', f
-      outer = this;
-    }
-  }));
-
-  owner.register('component:x-inner-in-template', Component.extend({
-    init() {
-      this._super(...arguments);
-      innerTemplate = this;
-    }
-  }));
-
-  owner.register('component:x-inner-in-layout', Component.extend({
-    init() {
-      this._super(...arguments);
-      innerLayout = this;
-    }
-  }));
-
-  owner.register('template:components/x-outer', compile('{{x-inner-in-layout}}{{yield}}'));
-
-  view = EmberView.extend({
-    [OWNER]: owner,
-    template: compile('{{#x-outer}}{{x-inner-in-template}}{{/x-outer}}')
-  }).create();
-
-  runAppend(view);
-
-  equal(innerTemplate.parentView, outer, 'receives the wrapping component as its parentView in template blocks');
-  equal(innerLayout.parentView, outer, 'receives the wrapping component as its parentView in layout');
-  equal(outer.parentView, view, 'x-outer receives the ambient scope as its parentView');
-});
-
-QUnit.test('newly-added sub-components get correct parentView', function() {
-  var outer, inner;
-
-  owner.register('component:x-outer', Component.extend({
-    init() {
-      this._super(...arguments);
-      outer = this;
-    }
-  }));
-
-  owner.register('component:x-inner', Component.extend({
-    init() {
-      this._super(...arguments);
-      inner = this;
-    }
-  }));
-
-  view = EmberView.extend({
-    [OWNER]: owner,
-    template: compile('{{#x-outer}}{{#if view.showInner}}{{x-inner}}{{/if}}{{/x-outer}}'),
-    showInner: false
-  }).create();
-
-  runAppend(view);
-
-  run(() => { view.set('showInner', true); });
-
-  equal(inner.parentView, outer, 'receives the wrapping component as its parentView in template blocks');
-  equal(outer.parentView, view, 'x-outer receives the ambient scope as its parentView');
-});
-
-QUnit.test('components should receive the viewRegistry from the parent view', function() {
-  var outer, innerTemplate, innerLayout;
-
-  var viewRegistry = {};
-
-  owner.register('component:x-outer', Component.extend({
-    init() {
-      this._super(...arguments);
-      outer = this;
-    }
-  }));
-
-  owner.register('component:x-inner-in-template', Component.extend({
-    init() {
-      this._super(...arguments);
-      innerTemplate = this;
-    }
-  }));
-
-  owner.register('component:x-inner-in-layout', Component.extend({
-    init() {
-      this._super(...arguments);
-      innerLayout = this;
-    }
-  }));
-
-  owner.register('template:components/x-outer', compile('{{x-inner-in-layout}}{{yield}}'));
-
-  view = EmberView.extend({
-    [OWNER]: owner,
-    _viewRegistry: viewRegistry,
-    template: compile('{{#x-outer}}{{x-inner-in-template}}{{/x-outer}}')
-  }).create();
-
-  runAppend(view);
-
-  equal(innerTemplate._viewRegistry, viewRegistry);
-  equal(innerLayout._viewRegistry, viewRegistry);
-  equal(outer._viewRegistry, viewRegistry);
-});
-
-QUnit.test('comopnent should rerender when a property is changed during children\'s rendering', function() {
```

Comment:
:heavy_check_mark: component should rerender when a property is changed during children\'s rendering